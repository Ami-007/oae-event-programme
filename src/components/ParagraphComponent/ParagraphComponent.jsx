import React from "react";
import "./ParagraphComponent.scss";
const ParagraphComponent = (props) => {
  const { text } = props;

  return (
    <article className="paragraph-component">
      <p>{text}</p>
    </article>
  );
};
export default ParagraphComponent;
