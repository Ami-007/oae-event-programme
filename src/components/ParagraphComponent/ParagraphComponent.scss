@use "../../assets/sass/variables.scss" as *;
@use "../../assets/sass/typography.scss" as *;

.paragraph-component {
  font-family: $modern-content;
  font-size: 1rem;
  padding: 0 41px;
}
