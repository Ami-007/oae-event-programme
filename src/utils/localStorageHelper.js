export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
