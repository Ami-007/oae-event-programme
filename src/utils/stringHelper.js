export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
