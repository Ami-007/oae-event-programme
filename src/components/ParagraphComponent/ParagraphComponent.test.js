import { screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import ParagraphComponent from "./ParagraphComponent.jsx";

it("should render the article", ()=> {
  customRender(<ParagraphComponent />);
  const article = screen.getByRole("article");

  expect(article).toMatchSnapshot();
})