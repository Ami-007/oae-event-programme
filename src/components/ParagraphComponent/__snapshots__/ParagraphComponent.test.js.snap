// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the article 1`] = `
<article
  class="paragraph-component"
>
  The Cantata being performed today is BWV 26 -
  <i>
    Ach wie flüchtig, ach wie nichtig.
  </i>
   The chorale of this name forms the linchpin of the outer movements and it is the only time Bach used this chorale in his cantatas. The melody of the chorale, written by Michael Franck in 1652, is used by others and the organ prelude today is a movement for manuals only (that is, no organ pedals!) from a set of variations on this tune by Georg Bohm (1661 - 1733), an organist whom J S Bach hugely admired and who may even have been his teacher. 
  <br />
  <br />
  The irresistable combination of three oboes brought a wealth of different repertoire to mind, but I have been enjoying digging into lesser-known works by George-Philipp Telemann this season and this movement from his “Darmstadt” Ouverture (the title of which roughly translates as “boasting”!) is a witty interplay between the wind and string teams.
</article>
`;
